src/index.js

```diff
@@ -1,52 +1,54 @@
 'use strict'
 
-var path = require('path')
+var checkPlatform = require('./checkPlatform')
 var goenv = require('go-platform')
-var version = require('./../package.json').version
-var request = require('request')
 var gunzip = require('gunzip-maybe')
+var path = require('path')
+var request = require('request')
 var tarFS = require('tar-fs')
+var unzip = require('unzip')
+var version = require('./../package.json').version
 
 module.exports = function (callback) {
   callback = callback || noop
-  checkPlatform(goenv) // make sure we can do this.
+
+  var fileExtension
+  var fileName
+  var fileStream
+  var installPath
+  var isWindows
+  var url
+
+  // make sure we can do this.
+  if (!checkPlatform.isSupportedArchitecture(goenv.GOARCH)) {
+    throw new Error('no binary available for arch: ' + goenv.GOARCH)
+  }
+
+  isWindows = checkPlatform.isWindows(goenv.GOOS)
 
   // hacky hack hack to work around unpublishability
   version = version.replace(/-[0-9]+/, '')
 
-  var filename = 'ipfs_v' + version + '_' + goenv.GOOS + '-' + goenv.GOARCH + '.tar.gz'
-  var url = 'http://dist.ipfs.io/go-ipfs/v' + version + '/go-' + filename
+  isWindows ? fileExtension = '.zip' : fileExtension = '.tar.gz'
 
-  var installPath = path.resolve(__dirname, '..')
+  fileName = 'ipfs_v' + version + '_' + goenv.GOOS + '-' + goenv.GOARCH + fileExtension
+  url = 'http://dist.ipfs.io/go-ipfs/v' + version + '/go-' + fileName
 
-  request
-    .get(url)
-    .pipe(gunzip())
+  installPath = path.resolve(__dirname, '..')
+
+  fileStream = request.get(url)
```

Comment:
Since fileStream doesn't change, let's do `const fileStream` instead.

Apologies for going super granular in review, I understand where the pattern of declaring vars at the top comes form, but in JS, things get declared when they are needed and now with ES6, it is expected that things that won't change its pointer be defined with const.